client/starwhale/api/_impl/dataset/loader.py

```diff
@@ -92,6 +94,42 @@ def _do_iter_row(self) -> t.Generator[TabularDatasetRow, None, None]:
             for row in self.tabular_dataset.scan():
                 yield row
         else:
+
+            def _travel_link(obj: t.Any) -> t.List[Link]:
+                _lks = []
+                if isinstance(obj, Link):
+                    _lks.append(obj)
+                elif isinstance(obj, dict):
+                    for v in obj.values():
+                        _lks.extend(_travel_link(v))
+                elif isinstance(obj, (list, tuple)):
+                    for v in obj:
+                        _lks.extend(_travel_link(v))
+                elif isinstance(obj, SwObject):
+                    for v in obj.__dict__.values():
+                        _lks.extend(_travel_link(v))
+                return _lks
+
+            def sign_uris(uris: t.List[str]) -> str:
+                r = (
+                    CloudRequestMixed()
+                    .do_http_request(
+                        f"/project/{self.dataset_uri.project}/{self.dataset_uri.object.typ}/{self.dataset_uri.object.name}/version/{self.dataset_uri.object.version}/sign-links",
+                        method=HTTPMethod.GET,
+                        instance_uri=self.dataset_uri,
+                        params={
```

Comment:
URL length may be too large for the server or proxy, pass data within body?

Reply:
done and tested locally
![image](https://user-images.githubusercontent.com/89630947/202350443-efce1157-8f70-48fb-9f76-ca6edab95a40.png)